OpenEdXMobile/src/main/java/org/edx/mobile/http/notifications/EdxCourseAccessErrorState.kt

```diff
@@ -0,0 +1,79 @@
+package org.edx.mobile.http.notifications
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.core.content.ContextCompat
+import org.edx.mobile.R
+import org.edx.mobile.databinding.LayoutCourseAccessErrorBinding
+import org.edx.mobile.extenstion.setVisibility
+
+class EdxCourseAccessErrorState @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private val layout: LayoutCourseAccessErrorBinding =
+        LayoutCourseAccessErrorBinding.inflate(LayoutInflater.from(context))
+
+    init {
+        this.addView(layout.root)
+        attrs?.let {
+            val typedArray = context.obtainStyledAttributes(it, R.styleable.StateLayout)
+            val state = State.values()[typedArray.getInt(R.styleable.StateLayout_state, 0)]
+            setState(state)
+        }
+    }
+
+    fun setState(state: State) {
+        when (state) {
+            State.AUDIT_ACCESS_EXPIRED -> {
+                layout.heading.text = context.getString(R.string.course_access_expired)
+                layout.description.text = context.getString(R.string.message_no_new_session)
+                layout.layoutUpgradeFeatures.root.setVisibility(false)
+                layout.primaryButton.shimmerViewContainer.hideShimmer()
+                layout.primaryButton.btnUpgrade.text = context.getText(R.string.label_find_a_course)
+                layout.primaryButton.btnUpgrade.icon = null
+                layout.secondaryButton.setVisibility(false)
+            }
+            State.IS_UPGRADEABLE -> {
+                layout.heading.text = context.getString(R.string.course_access_expired)
+                layout.description.text = context.getString(R.string.message_to_upgrade_course)
+                layout.layoutUpgradeFeatures.root.setVisibility(true)
+                layout.primaryButton.shimmerViewContainer.postDelayed({
+                    layout.primaryButton.btnUpgrade.isEnabled = false
+                    layout.primaryButton.shimmerViewContainer.startShimmer()
+                }, 250)
+                layout.secondaryButton.setVisibility(true)
+                layout.secondaryButton.text = context.getText(R.string.label_find_a_course)
+            }
+        }
+    }
+
+    fun setPrimaryButtonText(text: String) {
```

Comment:
better to rename it according to its functionality because right now it can only be used as upgradeable button

Reply:
retain the current method name `setPrimaryButtonText` cuz, within the state we are changing the Primary button test, and there is no keyword for upgrade button.